frontends/concrete-python/examples/xor_distance/hamming_distance.py

```diff
@@ -0,0 +1,228 @@
+    # fmt: off
+    hw_table_ref = [
+        0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3,
+        4, 3, 4, 4, 5, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4,
+        4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2,
+        3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5,
+        4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 1, 2, 2, 3, 2, 3, 3,
+        4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3,
+        3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5,
+        6, 6, 7, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6,
+        4, 5, 5, 6, 5, 6, 6, 7, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 4, 5, 5,
+        6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8]
+    # fmt: on
+
+    hw_table = fhe.LookupTable(hw_table_ref)
+    return hw_table[x]
+
+
+# Reference function for tests
+def dist_in_clear(x, y):
+    return np.sum(hw(x ^ y))
+
+
+# For all bitsize_w
+def dist_in_fhe_directly_from_cp(x, y):
+    return np.sum(hw(x ^ y))
+
+
+# For bitsize_w == 1 only
+def dist_in_fhe_with_bits_1b(x, y):
+    z = x + y
+    zx = fhe.bits(z)[0]
+    return np.sum(zx)
+
+
+# For all bitsize_w
+def dist_with_xor_internal(x, y, bitsize_w):
+    power = 2**bitsize_w
+    table = fhe.LookupTable([hw((i % power) ^ (i // power)) for i in range(power**2)])
+
+    z = x + power * y
+    zx = table[z]
+
+    return np.sum(zx)
+
+
+# Manage user args
+def manage_args():
+    parser = argparse.ArgumentParser(description="Hamming weight (aka XOR) distance in Concrete.")
+    parser.add_argument(
+        "--nb_bits",
+        dest="nb_bits",
+        action="store",
+        type=int,
+        default=120,
+        help="Number of bits (better to be a multiple of 12 to test all bitwidths)",
+    )
+    parser.add_argument(
+        "--shape",
+        dest="shape",
+        action="store",
+        type=int,
+        nargs="+",
+        default=None,
+        help="How to shape the bits. It has almost no importance for speed",
+    )
+    parser.add_argument(
+        "--show_mlir",
+        dest="show_mlir",
+        action="store_true",
+        help="Show the MLIR",
+    )
+    parser.add_argument(
+        "--repeat",
+        dest="repeat",
+        action="store",
+        type=int,
+        default=5,
+        help="Repeat x times",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    print()
+
+    # Options by the user
+    args = manage_args()
+
+    nb_bits = args.nb_bits
+    execution_times = {}
+
+    for bitsize_w in [1, 2, 3, 4]:
+        for algo in [
+            "dist_in_fhe_with_bits_1b",
+            "dist_in_fhe_with_xor_tables",
+            "dist_in_fhe_directly_from_cp",
+        ]:
+            # Because of B023 (Function definition does not bind loop variable) which is detected
```

Comment:
remove this comment ? if the code is valid (despite what ruff says), you can silent ruff.

Reply:
ruff is right so I have add to code in another way
but if you want, I remove the explanation
@rudy-6-4 : your call!

Reply:
I finally removed the comment.

Reply:
done